### PR TITLE
ROX-22480: Changelog entry for future hardening

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

This change announces a change of behaviour for some product endpoints.
This is a Changelog only modification.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,9 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
     - Secret, Secret Path
 - The following search terms will be disabled in the next release and removed from the secret context in 2 releases. They can be removed in the current release by setting ROX_SECRET_FILE_SEARCH=false:
   - Secret Type, Cert Expiration,Image Pull Secret Registry
+- The `/v1/availableAuthProviders` endpoint will soon require authentication and at least READ permission on the `Access` resource.
+  Ensure that any flow interacting with it is authenticated and has the proper permissions going forward.
+- The `/v1/tls-challenge` will soon require authentication, ensure that any flow interacting with these endpoints is authenticated going forward.
```
Comment (janisz) [inline CHANGELOG.md:34]: I'm not an expert but I think flow is not the word we are looking for
```suggestion
- The `/v1/tls-challenge` will soon require authentication. Ensure that all interactions with these endpoints include proper authentication
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -29,6 +29,9 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
     - Secret, Secret Path
 - The following search terms will be disabled in the next release and removed from the secret context in 2 releases. They can be removed in the current release by setting ROX_SECRET_FILE_SEARCH=false:
   - Secret Type, Cert Expiration,Image Pull Secret Registry
+- The `/v1/availableAuthProviders` endpoint will soon require authentication and at least READ permission on the `Access` resource.
+  Ensure that any flow interacting with it is authenticated and has the proper permissions going forward.
```
Comment (janisz) [inline CHANGELOG.md:33]: I think we prefer `future` over `soon` at least in this file.

File: CHANGELOG.md [hunk 3]
```diff
@@ -29,6 +29,9 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
     - Secret, Secret Path
 - The following search terms will be disabled in the next release and removed from the secret context in 2 releases. They can be removed in the current release by setting ROX_SECRET_FILE_SEARCH=false:
   - Secret Type, Cert Expiration,Image Pull Secret Registry
+- The `/v1/availableAuthProviders` endpoint will soon require authentication and at least READ permission on the `Access` resource.
```
Comment (dhaus67) [inline CHANGELOG.md:32]:
```suggestion
- The `/v1/availableAuthProviders` endpoint will in a future release require authentication and at least READ permission on the `Access` resource.
```

**Review by dhaus67 — COMMENTED** [review]
I suppose we plan to do this in 4.6? Might be worthwhile mentioning this in the Changelog directly (or the infamous "in two releases").

**Pull request merged**